dvc/data_cloud.py

```diff
@@ -73,13 +75,24 @@ def pull(self, cache, jobs=None, remote=None, show_checksums=False):
             show_checksums (bool): show checksums instead of file names in
                 information messages.
         """
-        return self.repo.cache.local.pull(
+        downloaded_items_num = self.repo.cache.local.pull(
             cache,
             jobs=jobs,
             remote=self.get_remote(remote, "pull"),
             show_checksums=show_checksums,
         )
 
+        if not verify:
+            self._save_pulled_checksums(cache)
+
+        return downloaded_items_num
+
+    def _save_pulled_checksums(self, cache):
+        for checksum in cache["local"].keys():
+            cache_file = self.repo.cache.local.checksum_to_path_info(checksum)
+            if self.repo.cache.local.exists(cache_file):
```

Comment:
We've talked about this being alright, as corrupted cache will be deleted prior to downloading (on `status`) and new corrupted files will stay in a temp file, so we are fine doing this, but let's put a comment to clarify that for anyone lurking around here in the future.